pkg/model/hvs/flavortemplate.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package hvs
+
+import (
+	"github.com/google/uuid"
+)
+
+type Meta struct {
+	FlavorPart  string `json:"flavor_part,omitempty"`
+	Source      string `json:"source,omitempty"`
+	Label       string `json:"label,omitempty"`
+	Vendor      string `json:"vendor,omitempty"`
+	IPAddress   string `json:"ip_address,omitempty"`
+	BiosName    string `json:"bios_name,omitempty"`
+	BiosVersion string `json:"bios_version,omitempty"`
+	OsName      string `json:"os_name,omitempty"`
+	OsVersion   string `json:"os_version,omitempty"`
+	VmmName     string `json:"vmm_name,omitempty"`
+	VmmVersion  string `json:"vmm_version,omitempty"`
+	TpmVersion  string `json:"tpm_version,omitempty"`
+	// swagger: strfmt uuid
+	HardwareUUID *uuid.UUID `json:"hardware_uuid,omitempty"`
+	Comment      string     `json:"comment,omitempty"`
+
+	TbootInstalled bool `json:"tboot_installed,omitempty"`
+	CBNTEnabled    bool `json:"cbnt_enabled,omitempty"`
+	UEFIEnabled    bool `json:"uefi_enabled,omitempty"`
+	SUEFIEnabled   bool `json:"suefi_enabled,omitempty"`
+	BMCEnabled     bool `json:"bmc_enabled,omitempty"`
+	PFREnabled     bool `json:"pfr_enabled,omitempty"`
+
+	DigestAlgorithm string `json:"digest_algorithm,omitempty"`
+}
+
+//PCR- Tp stpre PCR index with respective PCR bank.
+type PCR struct {
+	Index int    `json:"index"`
+	Bank  string `json:"bank"`
+}
+
+//EventLogEquals - To store event log need be equal with specified PCR.
+type EventLogEquals struct {
+	ExculdingTags *[]string `json:"excluding_tags"`
```

Comment:
I think we should avoid pointers in structures and not have pointers to arrays.  There is a need for optional json serialization (omitempty), but it also leads to a lot of nil checking or worse, the risk of panic if a nil check wasn't performed.

@ddhawal Do we have an standards around this?

Reply:
Initially we didn't used pointer, later we noticed in response if either excluding tags or eventlog includes  are not present then response will comes like

```
"eventlog_equals":null,
"eventlog_equals":null
```
So we included pointers to avoid this responses. Likewise we have to request like this to create FT, If we remove pointers .

```
"eventlog_equals": {},
"eventlog_includes": []
```

Reply:
slice are already pointers, not sure how defining pointer to slice would help here. Handle responses in functions if they are not appropriate instead of adding a pointer to the pointer